ansible/files/bootstrap/telemetry.sh

```diff
@@ -209,6 +224,52 @@ rm RRO-3.2.1-el6.x86_64.tar.gz
 cd RRO-3.2.1; sudo ./install.sh; cd $HOME
 $ANACONDA_PATH/bin/pip install rpy2
 
+# Setup metrics reporting
+SOPS_RPM=sops-3.0.4-1.x86_64.rpm
+SPARK_METRICS_CONF=/mnt/spark-metrics.properties
+touch $SPARK_METRICS_CONF
+if [ "$METRICS_PROVIDER" = "datadog" ] ; then
+    # Pull down and decrypt Datadog API key; see https://github.com/mozilla/emr-bootstrap-spark/pull/485
+    DD_DIR=$(mktemp -d)
+    aws s3 cp $TELEMETRY_CONF_BUCKET/packages/$SOPS_RPM $DD_DIR/
+    aws s3 cp $TELEMETRY_CONF_BUCKET/credentials/emr.yaml $DD_DIR/
+    sudo yum -y localinstall $DD_DIR/$SOPS_RPM
+    export DD_API_KEY=$(/usr/local/bin/sops --decrypt --extract '["datadog_agent::api_key"]' $DD_DIR/emr.yaml)
+    # Install datadog-agent; see https://app.datadoghq.com/account/settings#agent/aws
+    bash -c "$(curl -sLS https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"
+    rm -r "$DD_DIR"
+
+    # We opt not use to use https://docs.datadoghq.com/integrations/spark/
+    # Instead, we standardize on statsd metrics using Spark's StatsDSink
+    # and configure tags in the agent that will be applied by dogstatsd.
+    # This should make switching to a new metrics provider relatively straightforward.
+
+    MIN_EMR_LABEL=emr-5.13.0
+    EMR_LABEL=$(describe_cluster .Cluster.ReleaseLabel)
+    if [[ "$EMR_LABEL" =~ ^emr-5.* && "$MIN_EMR_LABEL" > "$EMR_LABEL" ]]; then
+        echo "ERROR: Incompatible options. StatsdSink is not present in $EMR_LABEL, but is required to support --metrics-provider=datadog; must specify $MIN_EMR_LABEL or later to enable metrics; terminating" 1>&2
+        # Cause the cluster to terminate with "Bootstrap failure"
+        exit 1
```

Comment:
I'm choosing to `exit 1` here to force the cluster to terminate. I tested this case and I'm able to access the logs to see what went wrong.

It seems better to force the cluster down during bootstrap than to let it come up and have the user be faced with mysterious errors when they try to run jobs.